### PR TITLE
feat: improved the nvim version check

### PR DESCRIPTION
**Pull request by xiyaowong**

### CODE REVIEW

**Review by theol0403 — APPROVED** [review]

File: src/main_controller.ts [hunk 1]
```diff
@@ -301,31 +301,51 @@ export class MainController implements vscode.Disposable {
...
+        const requirements = {
+            options: ["statuscolumn"],
+            functions: ["nvim_exec2"],
+        };
+        // check version
+        {
+            const [, info] = await this.client.apiInfo;
+            const { major, minor, patch } = info.version;
+            const currVersion = [major, minor, patch];
+            if (minVersion.some((v, i) => v > currVersion[i])) {
```
Comment (justinmk) [inline src/main_controller.ts:316]: This will reject version 0.10.1 if minVersion is 0.9.2

File: src/main_controller.ts [hunk 2]
```diff
@@ -301,31 +301,51 @@ export class MainController implements vscode.Disposable {
...
+                } else if (expr.startsWith("*")) {
+                    missFunctions.push(expr.substring(1));
+                }
+            }
+        });
+        const errMsgs = [
+            "Your nvim does not support the following features. Please check and update your nvim to the [latest stable version](https://github.com/neovim/neovim/releases/tag/stable). ",
+        ];
+        if (missOptions.length) errMsgs.push("Miss options: " + missOptions.join(", ") + ". ");
+        if (missFunctions.length) errMsgs.push("Miss functions: " + missFunctions.join(", ") + ". ");
```
Comment (justinmk) [inline src/main_controller.ts:346]: s/Miss/missing

Not sure this message is helpful though. It's a roundabout way of saying "you need version X".

File: src/main_controller.ts (hunk 1 above)
Comment (xiyaowong) [reply src/main_controller.ts:316]: My bad

File: src/main_controller.ts (hunk 2 above)
Comment (xiyaowong) [reply src/main_controller.ts:346]: It should be useful and users using the dev version can understand this information.
For example, the 0.9.0-dev version used in #1524

Comment (justinmk) [reply src/main_controller.ts:346]: > For example, the 0.9.0-dev version used in #1524

But `0.9.0-dev` should fail a check against 0.9.0. That's defined by semver.

Comment (xiyaowong) [reply src/main_controller.ts:346]: What if -dev contains the features we need?

Comment (justinmk) [reply src/main_controller.ts:346]: Why is it worth having complicated code to support that edge case?

Comment (xiyaowong) [reply src/main_controller.ts:346]: make sense

**Pull request merged**